addons/sale_stock/models/account_move.py

```diff
@@ -117,6 +117,9 @@ def _compute_delivery_date(self):
         # EXTENDS 'account'
         super()._compute_delivery_date()
         for move in self:
+            # Ensure the code isn't triggered by an override's dependencies
+            if move._origin.line_ids.sale_line_ids.order_id.ids == move.line_ids.sale_line_ids.order_id.ids:
+                continue
```

Comment:
Trying to bet on compute dependencies, `_origin` and stuff like that is often a bad idea

Considering that a legal requirement only matters if the invoice is posted*, the l10n override is more an help for the user rather than an important code. I mean, when posting the invoice, we will ensure that the field has a value:
https://github.com/odoo/odoo/blob/9f7128dd328254db1a2768b2af84ed61c2c7e599/addons/l10n_de/models/account_move.py#L22-L26
So we ensure the legal requirement will be respected, i.e.: that's the piece of code that really matters

Therefore, I would either convert the compute into an onchange (so we keep suggesting a value for the user) or delete it.

\* To be checked, even I'm 99% positive about this